### PR TITLE
[shopsys] move navigation feature from project-base to the packages

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| we have to move the functionality so we are able to properly fix the database schema after new version of `doctrine/dbal` was released (see https://github.com/doctrine/dbal/pull/6413)
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://rv-fix-build.odin.shopsys.cloud
  - https://cz.rv-fix-build.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php [hunk 1]
```diff
@@ -23,14 +23,14 @@
 class NavigationItemFormType extends AbstractType
 {
     /**
-     * @var \App\Model\Category\Category[]
+     * @var \Shopsys\FrameworkBundle\Model\Category\Category[]
      */
     private array $categoryPaths;
```
Comment (TomasLudvik) [inline packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php:28]: Should be protected, this applies to several occurences in this class

File: packages/framework/src/Model/Navigation/NavigationItemDataFactory.php [hunk 2]
```diff
@@ -2,29 +2,29 @@
...
 
 class NavigationItemDataFactory
 {
     /**
-     * @param \App\Model\Navigation\NavigationItemCategoryFacade $navigationItemCategoryFacade
+     * @param \Shopsys\FrameworkBundle\Model\Navigation\NavigationItemCategoryFacade $navigationItemCategoryFacade
      */
     public function __construct(
-        private NavigationItemCategoryFacade $navigationItemCategoryFacade,
+        protected NavigationItemCategoryFacade $navigationItemCategoryFacade,
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Navigation/NavigationItemDataFactory.php]: readonly?

File: packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
 
     /**
      * @param \Shopsys\FrameworkBundle\Form\Transformers\RemoveDuplicatesFromArrayTransformer $removeDuplicatesTransformer
      * @param \Shopsys\FrameworkBundle\Form\Transformers\CategoriesIdsToCategoriesTransformer $categoriesIdsToCategoriesTransformer
-     * @param \App\Model\Category\CategoryFacade $categoryFacade
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryFacade $categoryFacade
      */
     public function __construct(
         private RemoveDuplicatesFromArrayTransformer $removeDuplicatesTransformer,
```
Comment (TomasLudvik) [inline packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php]: readonly?

File: packages/framework/src/Model/Navigation/NavigationItemDetail.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    protected array $categoryDetailsByColumnNumber;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Navigation\NavigationItem $navigationItem
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category[][] $categoriesByColumnNumber
+     */
+    public function __construct(
+        protected NavigationItem $navigationItem,
+        array $categoriesByColumnNumber,
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Navigation/NavigationItemDetail.php]: Both could be protected and readonly right?

File: packages/framework/src/Model/Navigation/NavigationItemDetailFactory.php [hunk 5]
```diff
@@ -2,24 +2,24 @@
...
 
 class NavigationItemDetailFactory
 {
     /**
-     * @param \App\Model\Navigation\NavigationItemCategoryFacade $navigationItemCategoryFacade
+     * @param \Shopsys\FrameworkBundle\Model\Navigation\NavigationItemCategoryFacade $navigationItemCategoryFacade
      */
     public function __construct(
-        private NavigationItemCategoryFacade $navigationItemCategoryFacade,
+        protected NavigationItemCategoryFacade $navigationItemCategoryFacade,
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Navigation/NavigationItemDetailFactory.php]: readonly

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php (hunk 1 above)
Comment (vitek-rostislav) [reply packages/framework/src/Form/Admin/Navigation/NavigationItemFormType.php:28]: actually, this is a form type and these are not supposed to be extended by inheritance :slightly_smiling_face:

**Review by TomasLudvik — APPROVED** [review]

**Pull request merged**